app/controllers/shopify_app/app_proxy_controller.rb

```diff
@@ -0,0 +1,9 @@
+module ShopifyApp
+  class AppProxyController < ApplicationController
+    include ShopifyApp::AppProxyVerification
+
+    skip_before_action :verify_authenticity_token
+
+    before_action :verify_proxy_request
+  end
+end
```

Comment:
The way an app proxy works, it doesn't make sense to delegate to jobs like is done for webhooks. Webhooks in their nature are handled asynchronously - the app responds very quickly with a 200 OK, and then it is processed in a worker - whereas app proxy requests are just like any other normal HTTP request that the app receives. The only difference being that they come from Shopify with some extra query parameters, and we need to verify them before we continue processing the request.